### PR TITLE
Better markdown generation

### LINKED ISSUE #7: SemConv MD generator should not prepend "Conditional" to requirement conditions
From https://github.com/open-telemetry/opentelemetry-specification/pull/925#discussion_r484440799:

`"Required: Conditional No, but recommended"` looks odd and might be misleading.
The MD generator should only print the condition itself without prepending the word _Conditional_ in the table, unless it references a footnote.

### LINKED ISSUE #8: SemConv MD generator should write "or" between examples
From https://github.com/open-telemetry/opentelemetry-specification/pull/928#discussion_r485198089

Having an example per row makes it harder to understand that they are multiple examples and not a single one. The MD generator should use "or" instead of "<br>" to separate examples.

### LINKED ISSUE #9: SemConv MD generator should not use "enum" in attribute types
From https://github.com/open-telemetry/opentelemetry-specification/pull/926#discussion_r492054452

Attribute types that have a finite set of options, e.g. [network.transport](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/span-general.md#general-network-connection-attributes), should not report their type with the `enum` suffix.

### LINKED ISSUE #23: Generator should print examples between `[` `]` for array types

### PR DESCRIPTION
**Pull request by thisthat**
This PR addresses:
- Print "Conditional" only to footnotes - closes #7
- Write "or" between examples - closes #8 
- Don't print "enum" in attribute type - closes #9
- Print examples for arrays between `[]` - closes #23

### CODE REVIEW

**Comment by thisthat** [comment]
@justinfoote FYI

**Comment by Oberon00** [comment]
Can you reformat the PR description so that it auto-closes the issues? E.g. "Closes #7: Print..."

File: semantic-conventions/src/tests/data/markdown/deprecated/expected.md
Comment (Oberon00) [inline semantic-conventions/src/tests/data/markdown/deprecated/expected.md:7]:
```suggestion
| `http.method` | string | HTTP request method. | `GET`, `POST`, `HEAD` | Yes |
```

I would prefer commas. Using multiple `or`s sounds wrong.

**Review by Oberon00 — APPROVED** [review]

**Comment by Oberon00** [comment]
Thank you for taking care of these long-standing issues!

File: semantic-conventions/src/tests/data/markdown/deprecated/expected.md
Comment (thisthat) [reply semantic-conventions/src/tests/data/markdown/deprecated/expected.md:7]: On second thought, I agree. Reading or multiple times sounds weird.
Maybe `or` only for two options and `,` for multiple?

Comment (Oberon00) [reply semantic-conventions/src/tests/data/markdown/deprecated/expected.md:7]: If there is a concern that commas might make people think of arrays, I suggest: Use `or` only between the last two items of the list (like you normally do in English); or use semicolons instead of commas as separators.

File: semantic-conventions/src/tests/data/markdown/extend_constraint/expected.md
Comment (justinfoote) [inline semantic-conventions/src/tests/data/markdown/extend_constraint/expected.md:42]: 👏

File: semantic-conventions/src/tests/data/markdown/deprecated/expected.md
Comment (justinfoote) [reply semantic-conventions/src/tests/data/markdown/deprecated/expected.md:7]: I think putting a smart join in here (that uses `or` between two things, commas between more than two, and an additional `or` between the final two) seems like too much for this tool.  And I'd really prefer not to start a conversation about whether to use an oxford comma. 

I vote for semicolons. 😄

**Review by justinfoote — COMMENTED** [review]
🎉 This is a great addition! Replacing those `<br>`s makes these tables much more clear.

I'll approve once we have the comma/semicolon/or decision made.

File: semantic-conventions/src/tests/data/markdown/deprecated/expected.md
Comment (thisthat) [reply semantic-conventions/src/tests/data/markdown/deprecated/expected.md:7]: @open-telemetry/specs-approvers what would you suggest to use?

Comment (arminru) [reply semantic-conventions/src/tests/data/markdown/deprecated/expected.md:7]: The table header should say _Example**s**_ instead of _Example_ to make clear those are multiple, separate examples (still fine if there is only one example given).
I'd say commas and one _or_ (rather without oxford comma), just commas or just semicolons are all fine. When it says example**s** on top and also mentions the type, even commas should not confuse readers. Multiple *or*s really look odd, though.

Comment (Oberon00) [reply semantic-conventions/src/tests/data/markdown/deprecated/expected.md:7]: Let's use semicolons!

Comment (thisthat) [reply semantic-conventions/src/tests/data/markdown/deprecated/expected.md]: Done in 8053f628e25e7486766adb8a837d19ecb160a5cb

**Review by arminru — APPROVED** [review]
Thank you, Giovanni!

**Review by justinfoote — APPROVED** [review]

**Pull request merged**
